parquet/src/main/java/org/apache/iceberg/data/parquet/InternalWriter.java

```diff
@@ -40,11 +40,15 @@ public class InternalWriter<T extends StructLike> extends BaseParquetWriter<T> {
 
   private InternalWriter() {}
 
+  /**
+   * @deprecated will be removed in 1.10.0; use {@link #createWriter(Schema, MessageType)} instead.
+   */
+  @Deprecated
   public static <T extends StructLike> ParquetValueWriter<T> create(MessageType type) {
     return create((Types.StructType) null, type);
   }
 
-  public static <T extends StructLike> ParquetValueWriter<T> create(
+  public static <T extends StructLike> ParquetValueWriter<T> createWriter(
```

Comment:
This was needed because `InternalWriter::create` could refer to either `create(MessageType)` or `create(Schema, MessageType)`. Adding the ability to pass `BiFunction<Schema, MessageType, ParquetValueWriter<?>>` to `Parquet.createWriterFunc` caused compile failures because the calls were ambiguous.

To solve the problem, I renamed this method so it is unambiguous. This has not been in a release so it is safe. The `create(MessageType)` method has been in a release so it is deprecated for removal.

Reply:
I don't love this because we now have a confusing combination of `create`, `createWriter` and `buildWriter` floating around, but I also couldn't find a good alternative.

Reply:
I think we should just be more aggressive about fixing these. We should deprecate the old ones and support better names.